arch/mips/core/fatal.c

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2017 Imagination Technologies Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <kernel_structs.h>
+#include <inttypes.h>
+#include <misc/printk.h>
+
+const NANO_ESF _default_esf;
+
+/**
+ *
+ * @brief Fatal error handler
+ *
+ * This routine is called when a fatal error condition is detected by either
+ * hardware or software.
+ *
+ * The caller is expected to always provide a usable ESF.  In the event that the
+ * fatal error does not have a hardware generated ESF, the caller should either
+ * create its own or call _Fault instead.
+ *
+ * @param reason the reason that the handler was called
+ * @param esf pointer to the exception stack frame
+ *
+ * @return This function does not return.
+ */
+FUNC_NORETURN void _NanoFatalErrorHandler(unsigned int reason,
```

Comment:
this isn't an acceptable implementation.
_NanoFatalErrorHandler prints an error message and then calls _SysFatalErrorHandler, which implements policy on what to do when a thread encounters an exception.